RetailCoder.VBE/Inspections/QuickFixes/AssignedByValParameterMakeLocalCopyQuickFix.cs

```diff
@@ -97,13 +97,42 @@ private void ReplaceAssignedByValParameterReferences(IModuleRewriter rewriter, s
             }
         }
 
+
         private void InsertLocalVariableDeclarationAndAssignment(IModuleRewriter rewriter, string localIdentifier)
         {
-            var content = Tokens.Dim + " " + localIdentifier + " " + Tokens.As + " " + _target.AsTypeName + Environment.NewLine
-                + (_target.AsTypeDeclaration is ClassModuleDeclaration ? Tokens.Set + " " : string.Empty)
-                + localIdentifier + " = " + _target.IdentifierName;
+            var content = Tokens.Dim + " " + localIdentifier + " " + Tokens.As + " " + _target.AsTypeName + Environment.NewLine;
+            if (IsBaseTypeContext(_target))
+            {
+                content = content + localIdentifier + " = " + _target.IdentifierName;
+            }
+            else 
+            {
+                //All we can know is that it is not a Base type.  Let VBA determine
+                //the right way to assign the parameter.  The user can simplify it later.
+                string insertIsObjectCheck =
+@"If(IsObject({1})) Then
```

Comment:
A better example than using a Range would be:
```
Public Sub Foo(ByVal arg1)
   arg1 = new Collection
End Sub
```
The Range example cited earlier actually compiles and runs...the above example generates a runtime error - and is not detected the by the ObjectVariableNotSet inspection.  I didn't spot an inspection for anything like 'ParameterNotTyped' - though maybe it could be useful.